stumpy/search.py

```diff
@@ -0,0 +1,361 @@
+            "Multidimensional motif discovery is not yet supported."
+        )
+
+    if P.ndim != 1:  # pragma: no cover
+        raise ValueError(
+            f"T is {P.ndim}-dimensional and must be 1-dimensional. "
+            "Multidimensional motif discovery is not yet supported."
+        )
+
+    T = T.copy()
+    T = T[np.newaxis, :]
+
+    P = P.copy().astype(float)
+    P = P[np.newaxis, :]
+
+    S = np.zeros((1, P.shape[1]), dtype=float)
+
+    dimensions = [0]
+    m = T.shape[1] - P.shape[1] + 1
+
+    if excl_zone is None:  # pragma: no cover
+        excl_zone = int(np.ceil(m / 4))
+    if atol is None:  # pragma: no cover
+        atol = 0.25 * np.std(P)
+    if max_occurrences is None:  # pragma: no cover
+        max_occurrences = np.inf
+
+    T, M_T, Σ_T = core.preprocess(T, m)
+
+    result = _search(
+        T=T,
+        P=P,
+        S=S,
+        dimensions=dimensions,
+        k=k,
+        M_T=M_T,
+        Σ_T=Σ_T,
+        excl_zone=excl_zone,
+        min_neighbors=min_neighbors,
+        max_occurrences=max_occurrences,
+        atol=atol,
+        rtol=rtol,
+    )
+    return result
+
+
+def search_k_discords(T, P, k=1, excl_zone=None):
+    """Find the top k discords of the time series `T`.
+
+    Parameters
+    ----------
+    T : ndarray
+        The time series of interest
+
+    P : ndarray
+        Matrix Profile of `T` (result of a self-join)
+
+    k : int
+        Number of discords to search. Defaults to `1`.
+
+    excl_zone : int or None
+        Size of the exclusion zone.
+        If `None`, defaults to `m/4`, where `m` is the length of `Q`.
+
+    Return
+    ------
+    top_motif_indices : list
+        Each item is another list, with each item representing an occurrence of the
+        motif in `T`, sorted by distance from the "original occurrence", i.e. the
+        subsequence of `T` starting at the index in the first place of the list
+
+    top_motif_values : list
+        The matrix profile values of the "original occurrences"
+    """
+    if k < 1:  # pragma: no cover
+        raise ValueError("k (the number of motifs) has to be at least 1")
+
+    if T.ndim != 1:  # pragma: no cover
+        raise ValueError(
+            f"T is {T.ndim}-dimensional and must be 1-dimensional. "
+            "Multidimensional motif discovery is not yet supported."
+        )
+
+    if P.ndim != 1:  # pragma: no cover
+        raise ValueError(
+            f"T is {P.ndim}-dimensional and must be 1-dimensional. "
+            "Multidimensional motif discovery is not yet supported."
+        )
+
+    T = T.copy()
+    T = T[np.newaxis, :]
+
+    P = P.copy().astype(float)
+    P = np.max(P) - P
+    P = P[np.newaxis, :]
+
+    S = np.zeros((1, P.shape[1]), dtype=float)
+
+    dimensions = [0]
+    m = T.shape[1] - P.shape[1] + 1
+
+    if excl_zone is None:  # pragma: no cover
+        excl_zone = int(np.ceil(m / 4))
+
+    T, M_T, Σ_T = core.preprocess(T, m)
+
+    result = _search(
```

Comment:
Is it correct that things like `min_neighbors`, `max_occurrences`, `atol`, and `rtol` are different for discords than motifs?